src/lxml/tests/test_etree.py

```diff
@@ -4389,6 +4420,16 @@ def test_xinclude_failures(self):
         self.assertEqual(str(cm.exception),
                          "recursive include of 'Recursive2.xml' detected")
 
+    def test_multiple_include_of_same_file(self):
+        # Test that including the same file multiple times, but on the same level
+        # is not detected as recursive include
+        document = self.xinclude_loader("NonRecursive2.xml").getroottree()
+        self.include(document, self.xinclude_loader)
+
+        # same but for more than one level
+        document = self.xinclude_loader("NonRecursive1.xml").getroottree()
+        self.include(document, self.xinclude_loader)
```

Comment:
I think another good test case would be to include the leaf file _only_ from the second level include files and not from the top-level file at all.

Reply:
added proposed testcase